tests/unit/test_footprinter.cc

```diff
@@ -474,48 +515,21 @@ TEST_F(TestFootprinter, test_vector) {
       sizeof(v)
     );
   }
-}
-
-struct TestBase {
-
-  checkpoint_virtual_serialize_root()
-
-  virtual ~TestBase() = default;
-
-  template <typename SerializerT>
-  void serialize(SerializerT& s) {
-    s | a;
-  }
-
-private:
-  int a = 0;
-};
-
-struct TestDerived2 : TestBase {
-  explicit TestDerived2(int i) {}
-  explicit TestDerived2(SERIALIZE_CONSTRUCT_TAG) {}
 
-  checkpoint_virtual_serialize_derived_from(TestBase)
+  {
+    std::vector<TestDerived2> v;
+    EXPECT_GE(
+      checkpoint::getMemoryFootprint(v),
+      sizeof(v)
+    );
 
-  template <
-    typename SerializerT,
-    typename = std::enable_if_t<
-      std::is_same<
-        SerializerT,
-        checkpoint::Footprinter
-      >::value
-    >
-  >
-  void serialize(SerializerT& s) {
-    s | raw_pointer;
-    s | shared_pointer;
+    v.reserve(3);
+    EXPECT_EQ(
```

Comment:
I think this case also needs to be `EXPECT_GE` though - there's no requirement that the memory allocated by `reserve` be a 'tight fit'

Reply:
you beat me by 90 seconds with this comment :)

Reply:
in the end, it will always work because the comparison uses `v.capacity()` (only `reserve` uses magic number)

Reply:
I see, great.